test/test_blocks.py

```diff
@@ -1412,6 +1412,53 @@ def moo(a: gr.Request, b, c: int, d: gr.Request):
         inputs_ = gr.helpers.special_args(moo, copy.deepcopy(inputs), request)[0]
         assert inputs_ == [request] + inputs + [request]
 
+    def test_default_args(self):
```

Comment:
Awesome test! Mind adding some cases where moo takes `gr: Request` and `gr.SelectData` for completeness?